src/qibo/abstractions/gates.py

```diff
@@ -1550,3 +1550,33 @@ def __init__(self, q, t1, t2, time, excited_population=0, seed=None):
             seed=seed)
         # this case can only be applied to density matrices
         self.density_matrix = True
+
+
+class FusedGate(Gate):
```

Comment:
Maybe we could add a comment or docstring regarding how this class is used in the code and what are its limitations (e.g. if it would generalize to n-qubit fusion.)